src/Uno.Toolkit.UI/Controls/Shadows/ShadowContainer.Paint.cs

```diff
@@ -0,0 +1,368 @@
+		if (Cache.TryGetValue(shadowsKey, out var shadowsImage))
+		{
+			if (pixelRatioChanged)
+			{
+				// Monitor pixel density changed, need to remove cached image
+				Cache.Remove(shadowsKey);
+			}
+			else
+			{
+				canvas.DrawImage(shadowsImage, SKPoint.Empty);
+				canvas.Restore();
+				return;
+			}
+		}
+
+		float childWidth = (float)width * pixelRatio;
+		float childHeight = (float)height * pixelRatio;
+
+		float diffWidthSurfaceChild = surfaceWidth - childWidth;
+		float diffHeightSurfaceChild = surfaceHeight - childHeight;
+		canvas.Translate(diffWidthSurfaceChild / 2, diffHeightSurfaceChild / 2);
+
+		using var paint = new SKPaint();
+		paint.IsAntialias = true;
+
+		float cornerRadius = (float)_cornerRadius.BottomRight * pixelRatio;
+
+		foreach (var shadow in shadows.Where(s => !s.IsInner))
+		{
+			var skShadow = SKShadow.From(shadow, childWidth, childHeight, cornerRadius, pixelRatio);
+
+			DrawDropShadow(canvas, paint, skShadow);
+		}
+
+		// Always draw inner shadows on top of the drop shadows
+		if (hasInnerShadow)
+		{
+			var contentShape = new SKRoundRect(new SKRect(0, 0, childWidth, childHeight), cornerRadius);
+			canvas.ClipRoundRect(contentShape, antialias: true);
+
+			// Draw the content background first
+			if (_currentContentBackgroundColor.HasValue)
+			{
+				var contentBackgroundColor = ToSkiaColor(_currentContentBackgroundColor.Value);
+				DrawContentBackground(canvas, contentBackgroundColor, contentShape);
+			}
+
+			// Then we draw the inner shadows
+			foreach (var shadow in shadows.Where(s => s.IsInner))
+			{
+				var skShadow = SKShadow.From(shadow, childWidth, childHeight, cornerRadius, pixelRatio);
+
+				DrawInnerShadow(canvas, paint, skShadow);
+			}
+		}
+
+		canvas.Restore();
+
+		if (!_shadowPropertyChanged)
+		{
+			// If a property has changed dynamically, we don't want to cache the updated shadows
+			Cache.AddOrUpdate(shadowsKey, surface.Snapshot());
+		}
+
+		_shadowPropertyChanged = false;
+	}
+
+	private bool ProcessContentBackgroundIfNeeded(out Color? contentBackgroundColor)
+	{
+		contentBackgroundColor = null;
+		if (TryGetContentBackground(out var background))
+		{
+			if (background is not SolidColorBrush backgroundColorBrush)
+			{
+				throw new NotSupportedException("[ShadowContainer] Unsupported Background brush: when using inner shadows the only supported brush type for the Background property is SolidBrushColor");
+			}
+
+			if (backgroundColorBrush.Color != Color.FromArgb(0, 0, 0, 0))
+			{
+				contentBackgroundColor = backgroundColorBrush.Color;
+			}
+
+			TrySetContentBackground(new SolidColorBrush(Color.FromArgb(0, 0, 0, 0)));
+			return true;
+		}
+
+		return false;
+	}
+
+	private static void DrawContentBackground(SKCanvas canvas, SKColor contentBackgroundColor, SKRoundRect childShape)
+	{
+		using var backgroundPaint = new SKPaint
+		{
+			Color = contentBackgroundColor,
+			Style = SKPaintStyle.Fill,
+		};
+
+		System.Diagnostics.Debug.WriteLine(
```

Comment:
we are using Microsoft.Extensions.Logging in pretty much most places
convert those to something shown like here:  https://github.com/unoplatform/uno.toolkit.ui/blob/3.0.0/src/Uno.Toolkit.UI/Behaviors/InputExtensions.cs#L130-L133